org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/FormatterHandler.java

```diff
@@ -311,4 +312,39 @@ private IRegion getRegion(ICompilationUnit cu, IDocument document, Position posi
 		return null;
 	}
 
+	/**
+	 * A utility function to format a String with given formatter options.
+	 *
+	 * @see org.eclipse.jdt.core.formatter.CodeFormatterApplication#formatFile(File,
+	 *      CodeFormatter)
+	 * @param content
+	 *                    the content to format
+	 * @param options
+	 *                    the Map includes formatter options

```

Comment:
the Map includes formatter options. If set to <code>null</code>, then use the current settings from <code>JavaCore#getOptions</code>.